dspace-api/src/main/java/org/dspace/authorize/AuthorizeConfiguration.java

```diff
@@ -16,172 +17,35 @@
-            "core.authorization.collection-admin.template-item", true);
-
-    private static boolean can_collectionAdmin_submitters = ConfigurationManager
-        .getBooleanProperty(
-            "core.authorization.collection-admin.submitters", true);
-
-    private static boolean can_collectionAdmin_workflows = ConfigurationManager
-        .getBooleanProperty(
-            "core.authorization.collection-admin.workflows", true);
-
-    private static boolean can_collectionAdmin_adminGroup = ConfigurationManager
-        .getBooleanProperty(
-            "core.authorization.collection-admin.admin-group", true);
-
-    private static boolean can_collectionAdmin_itemDelete = ConfigurationManager
-        .getBooleanProperty(
-            "core.authorization.collection-admin.item.delete", true);
-
-    private static boolean can_collectionAdmin_itemWithdraw = ConfigurationManager
-        .getBooleanProperty(
-            "core.authorization.collection-admin.item.withdraw", true);
-
-    private static boolean can_collectionAdmin_itemReinstatiate = ConfigurationManager
-        .getBooleanProperty(
-            "core.authorization.collection-admin.item.reinstatiate",
-            true);
-
-    private static boolean can_collectionAdmin_itemPolicies = ConfigurationManager
-        .getBooleanProperty(
-            "core.authorization.collection-admin.item.policies", true);
-
-    // # also bundle
-    private static boolean can_collectionAdmin_itemCreateBitstream = ConfigurationManager
-        .getBooleanProperty(
-            "core.authorization.collection-admin.item.create-bitstream",
-            true);
-
-    private static boolean can_collectionAdmin_itemDeleteBitstream = ConfigurationManager
-        .getBooleanProperty(
-            "core.authorization.collection-admin.item.delete-bitstream",
-            true);
-
-    private static boolean can_collectionAdmin_itemAdminccLicense = ConfigurationManager
-        .getBooleanProperty(
-            "core.authorization.collection-admin.item-admin.cc-license",
-            true);
-
-    // # ITEM ADMIN
-    private static boolean can_itemAdmin_policies = ConfigurationManager
-        .getBooleanProperty("core.authorization.item-admin.policies", true);
-
-    // # also bundle
-    private static boolean can_itemAdmin_createBitstream = ConfigurationManager
-        .getBooleanProperty(
-            "core.authorization.item-admin.create-bitstream", true);
-
-    private static boolean can_itemAdmin_deleteBitstream = ConfigurationManager
-        .getBooleanProperty(
-            "core.authorization.item-admin.delete-bitstream", true);
-
-    private static boolean can_itemAdmin_ccLicense = ConfigurationManager
-        .getBooleanProperty("core.authorization.item-admin.cc-license",
-                            true);
+    /**
+     * A static reference to the {@link ConfigurationService} see the init method for initialization
+     */
+    private static ConfigurationService configurationService;
```

Comment:
A local variable would work too, since a method invocation shouldn't last long enough to worry about someone changing the configuration.  This could help readability, and avoid exhausting the reader (or the author!) with repeated long expressions if the method uses more than one configuration property.

Reply:
@mwoodiupui : yes, agreed for most classes/scenarios.  A local variable or `@Autowired` would normally be the best approach to getting the Configuration & working with it.  

However this `AuthorizeConfiguration` class seems to be a Util style class...it's a ton of static methods that are one-line requests for a specific config value.  So, in this class, the above DSpaceServiceFactory one-liner is probably the best we can do.

Reply:
Hi @tdonohue I agree with you that this class is not ideal and it could nice to get ride of it in future I have avoided to do that to avoid to broad the scope of this PR and make it bigger than needed. The only change that I have had to introduce is the switch from the ConfigurationManager to the ConfigurationService to be able to easily change the configuration in the test. That said, the static reference to the ConfigurationService is not a problem for the reloading as this happen internally to the ConfigurationService itself. The issue would be to cache an actual value otherwise, it would mean that when a config file is changed the whole spring context is refreshed and this doesn't look to be the case. Anyway, I have introduced an IntegrationTest in the dspace-api to specifically test that see 
https://github.com/DSpace/DSpace/pull/2663/files#diff-021a0801128069441947b7de3acc859d

BTW I haven't found test related to the reloadable configuration if you are aware of such test please give me direction maybe the extra methods added in the AbstractIntegrationTest 
https://github.com/DSpace/DSpace/pull/2663/files#diff-1be4c4783e30c75d0db54c3b3b049bfd
already exists elsewhere, otherwise it will be good have now a first test in place that touch also that aspect

Reply:
@abollini : Tests for reloading configs exist in the `dspace-services` layer.  Here's the one that tests for an automatic reload. There's others in this same Test class that test other areas of configuration reload/population, etc. https://github.com/DSpace/DSpace/blob/master/dspace-services/src/test/java/org/dspace/servicemanager/config/DSpaceConfigurationServiceTest.java#L529

I still have a strong preference for avoiding any static references to cbeans/classes.  It's really unnecessary here, since you can get a reference via the `DSpaceServicesFactory` rather than re-caching it locally.

That said, I'll let it slide here since you've added tests to prove that it works as expected, regardless of the static reference.